While GCC does not understand, that `static constexpr` data members could be treated just like global constexpr variables, i.e. at compile time, MSVC apparently does not understand `constexpr` functions:
```
D:\a\alpaka\alpaka\include\alpaka/block/shared/dyn/BlockSharedMemDynMember.hpp(85): error C2975: '_Size': invalid template argument for 'std::array', expected compile-time constant expression [D:\a\alpaka\alpaka\build\test\unit\math\math.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\include\array(20): note: see declaration of '_Size'
  D:\a\alpaka\alpaka\include\alpaka/block/shared/dyn/BlockSharedMemDynMember.hpp(87): note: see reference to class template instantiation 
```
@BenjaminW3 Do you know a fix for this?

--- a/include/alpaka/acc/AccCpuOmp2Blocks.hpp
+++ b/include/alpaka/acc/AccCpuOmp2Blocks.hpp
@@ -184,7 +184,7 @@ namespace alpaka
                         // m_threadElemCountMax
                         std::numeric_limits<TIdx>::max(),
                         // m_sharedMemSizeBytes
-                        static_cast<size_t>( acc::AccCpuOmp2Blocks<TDim, TIdx>::staticAllocBytes )};
+                        static_cast<size_t>( acc::AccCpuOmp2Blocks<TDim, TIdx>::staticAllocBytes() )};
                 }
             };
             //#############################################################################

--- a/include/alpaka/acc/AccCpuSerial.hpp
+++ b/include/alpaka/acc/AccCpuSerial.hpp
@@ -178,7 +178,7 @@ namespace alpaka
                         // m_threadElemCountMax
                         std::numeric_limits<TIdx>::max(),
                         // m_sharedMemSizeBytes
-                        static_cast< size_t >( acc::AccCpuSerial<TDim, TIdx>::staticAllocBytes )};
+                        static_cast< size_t >( acc::AccCpuSerial<TDim, TIdx>::staticAllocBytes() )};
                 }
             };
             //#############################################################################

--- a/include/alpaka/acc/AccCpuTbbBlocks.hpp
+++ b/include/alpaka/acc/AccCpuTbbBlocks.hpp
@@ -176,7 +176,7 @@ namespace alpaka
                         // m_threadElemCountMax
                         std::numeric_limits<TIdx>::max(),
                         // m_sharedMemSizeBytes
-                        static_cast< size_t >( acc::AccCpuTbbBlocks<TDim, TIdx>::staticAllocBytes )};
+                        static_cast< size_t >( acc::AccCpuTbbBlocks<TDim, TIdx>::staticAllocBytes() )};
                 }
 
             };

--- a/include/alpaka/block/shared/dyn/BlockSharedMemDynMember.hpp
+++ b/include/alpaka/block/shared/dyn/BlockSharedMemDynMember.hpp
@@ -28,6 +28,20 @@ namespace alpaka
         {
             namespace dyn
             {
+                namespace detail
+                {
+                    //#############################################################################
+                    //! "namespace" for static constexpr members that should be in BlockSharedMemDynMember
+                    //! but cannot be because having a static const member breaks GCC 10
+                    //! OpenMP target and OpenACC: type not mappable.
+                    template<unsigned int TStaticAllocKiB>
+                    struct BlockSharedMemDynMemberStatic
+                    {
+                        //! Storage size in bytes
+                        static constexpr unsigned int staticAllocBytes = TStaticAllocKiB<<10;
+                    };
+                }
+
 #if BOOST_COMP_MSVC || defined(BOOST_COMP_MSVC_EMULATED)
     #pragma warning(push)
     #pragma warning(disable: 4324)  // warning C4324: structure was padded due to alignment specifier
@@ -47,7 +61,7 @@ namespace alpaka
                              + (sizeBytes%core::vectorization::defaultAlignment>0))*core::vectorization::defaultAlignment)
                     {
 #if (defined ALPAKA_DEBUG_OFFLOAD_ASSUME_HOST) && (! defined NDEBUG)
-                        ALPAKA_ASSERT(sizeBytes <= staticAllocBytes);
+                        ALPAKA_ASSERT(sizeBytes <= staticAllocBytes());
 #endif
                     }
                     //-----------------------------------------------------------------------------
@@ -74,14 +88,15 @@ namespace alpaka
                      */
                     unsigned int staticMemCapacity() const
                     {
-                        return staticAllocBytes - m_dynPitch;
+                        return staticAllocBytes() - m_dynPitch;
                     }
 
                     //! Storage size in bytes
-                    static constexpr unsigned int staticAllocBytes = TStaticAllocKiB<<10;
+                    static constexpr unsigned int staticAllocBytes() {return detail::BlockSharedMemDynMemberStatic<TStaticAllocKiB>::staticAllocBytes;}
 
                 private:
-                    mutable std::array<uint8_t, staticAllocBytes> m_mem;
+
+                    mutable std::array<uint8_t, detail::BlockSharedMemDynMemberStatic<TStaticAllocKiB>::staticAllocBytes> m_mem;
                     unsigned int m_dynPitch;
                 };
 #if BOOST_COMP_MSVC || defined(BOOST_COMP_MSVC_EMULATED)

